packages/pages/src/common/src/project/structure.ts

```diff
@@ -112,69 +155,59 @@ export type Optional<T> = {
  * @public
  */
 export class ProjectStructure {
-  #config: ProjectStructureConfig;
-
-  sitesConfigRoot: Path;
-  templatesRoot: Path;
-  serverlessFunctionsRoot: Path;
-  scope?: string;
-  scopedTemplatesPath?: Path;
-  scopedSitesConfigPath?: Path;
-
-  distRoot: Path;
-  serverBundleOutputRoot: Path;
-  renderBundleOutputRoot: Path;
-  functionBundleOutputRoot: Path;
-  ciConfig: string;
-  featuresConfig: string;
-  envVarDir: string;
-  envVarPrefix: string;
-  siteStreamConfig: string;
+  config: ProjectStructureConfig;
 
   constructor(config?: Optional<ProjectStructureConfig>) {
-    this.#config = merge(defaultProjectStructureConfig, config);
-    this.sitesConfigRoot = new Path(
-      this.#config.filepathsConfig.sitesConfigRoot
-    );
-    this.templatesRoot = new Path(this.#config.filepathsConfig.templatesRoot);
-    this.serverlessFunctionsRoot = new Path(
-      this.#config.filepathsConfig.functionsRoot
+    const mergedConfig = merge(defaultProjectStructureConfig, config);
+    this.config = mergedConfig;
+  }
+
+  static init = async (
```

Comment:
Won't we need some validation here to ensure that these paths don't conflict with each other? Are relative paths allowed in the config options? if so you could probably get into some weird combinations.

Reply:
Hmm, that's an interesting point. I've been tempted for awhile to just remove this config as a plugin option anyways. When would people ever use it? Maybe in a future world of custom stuff or trying to support some other framework? Validation will be necessary if we keep it (or even just for the assetDir you can configure), but it will be pretty annoying. Check that literally no field overlaps or matches and there are no `\`. I will add validation in another PR.